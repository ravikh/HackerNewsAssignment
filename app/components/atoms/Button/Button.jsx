import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import styles from './Button.style';

const Button = ({ children, showAsLink, disabled, isPrimary, isSecondary, ariaLabel, className, type, hideText, onClick }) => {
    return (
        <button
            aria-label={ariaLabel}
            type={type}
            className={className}
            disabled={disabled}
            onClick={onClick}
            >
            {children}
        </button>
    )
}

Button.defaultProps = {
    className: '',
    ariaLabel: '',
    showAsLink: false,
    disabled: false,
    isPrimary: true,
    isSecondary: false,
    hideText: false,
    onClick: () => {}
};

Button.propTypes = {
    type: PropTypes.string,
    className: PropTypes.string,
    showAsLink: PropTypes.bool,
    disabled: PropTypes.bool,
    isPrimary: PropTypes.bool,
    isSecondary: PropTypes.bool,
    hideText: PropTypes.bool,
    onClick: PropTypes.func
};

const StyledButton = styled(Button)`
  ${styles};
`;

export default StyledButton;