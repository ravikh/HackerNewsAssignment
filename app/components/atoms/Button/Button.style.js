
import {css} from 'styled-components';

const ButtonStyles = css`
    border: none;
    border-radius: 0;
    position: relative;
    min-height: 2rem;
    color: ${props => props.theme.colors.primary.dark};
    cursor: ${(props) => (props.disabled ? 'not-allowed' : 'pointer')};

    ${(props) => props.showAsLink ? `
        background: none;
        min-height: 1rem;
        color: ${props.theme.colors.common.blue};
      `
        : ``};

    ${(props) => (props.hideText && 
        `
            text-indent: -999rem;
        `)}
`;

export default ButtonStyles;