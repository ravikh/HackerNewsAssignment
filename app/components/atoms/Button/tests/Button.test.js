import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Button from '../Button';

import Theme from '../../../../theme';

describe('Button', () => {
  let button = '';
  const theme = { ...Theme };

  test('should render correctly', () => {
    button = shallow(<Button theme={theme} type="button"/>);
    expect(button).toMatchSnapshot();
  });
});