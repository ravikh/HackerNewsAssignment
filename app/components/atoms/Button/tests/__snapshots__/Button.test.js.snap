// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<Button
  ariaLabel=""
  className=""
  disabled={false}
  hideText={false}
  isPrimary={true}
  isSecondary={false}
  onClick={[Function]}
  showAsLink={false}
  theme={
    Object {
      "colors": Object {
        "background": Object {
          "default": "var(--primary-bg-color)",
          "errorMessage": "var(--white)",
          "header": "var(--primary-color)",
        },
        "common": Object {
          "black": "var(--black)",
          "blue": "var(--blue)",
          "white": "var(--white)",
        },
        "error": Object {
          "primary": "var(--red)",
          "secondary": "var(--grey)",
        },
        "primary": Object {
          "dark": "var(--black)",
          "light": "var(--grey)",
        },
        "secondary": Object {
          "dark": "var(--black)",
          "light": "var(--grey)",
        },
        "text": Object {
          "disabled": "var(--text-secondary)",
          "primary": "var(--text-primary)",
          "secondary": "var(--text-secondary)",
        },
      },
      "typography": Object {
        "font_default": "
        font-family: var(--font-family-primary);
        font-size: var(--base-font);
        line-height: var(--line-height-normal);
    ",
        "font_h1": "
        font-family: var(--font-family-primary);
        font-size: var(--font-size-h1);
        line-height: var(--line-height-h1);
    ",
        "font_h2": "
        font-family: var(--font-family-primary);
        font-size: var(--font-size-h2);
        line-height: var(--line-height-h2);
    ",
        "font_h3": "
        font-family: var(--font-family-primary);
        font-size: var(--font-size-h3);
        line-height: var(--line-height-h3);
    ",
        "font_h4": "
        font-family: var(--font-family-primary);
        font-size: var(--font-size-h4);
        line-height: var(--line-height-h5);
    ",
        "font_link": "
        font-family: var(--font-family-primary);
        font-size: var(--font-size-link);
        line-height: var(--line-height-link);
    ",
      },
    }
  }
  type="button"
/>
`;
